### DISCUSSION TITLE
Allow re-mapping of ⎋ Escape

### DISCUSSION THREAD

**Discussion by d34dh0r53** [blinksh/blink#1579] [Support]
### Checklist

- [x] Using latest version on the [App Store](http://itunes.apple.com/app/id1156707581)
- [x] Read the [docs](http://blink.sh/docs/)
- [x] Searched for [existing GitHub issues](https://github.com/blinksh/blink/issues)

### Configuration

- Blink Version: `v15.0.8.539`
- iPadOS Version: `16.0 (20A5349b)`

### Describe the ~~bug~~ feature

I use the Apple Magic keyboard with my iPad.  In iPadOS I’ve remapped `caps` to `⎋ Escape` which works great, but as a VIM user it would be awesome to have the ability to use `caps` as either `⎋ Escape` on KeyDown/KeyUp or as `ctrl` on KeyDown + Key.  It looks like that ability is present if I leave the key as `caps` in iPadOS, can we add `⎋ Escape` as a re-mappable key?

**Comment by carloscabanero** [comment]
If I understand you correctly, this should be already doable with Blink. I would take a different direction though, let the system handle the Modifier (Ctrl), while Blink handles the Press (ESC). First, configure on iPadOS to have CAPS as the Ctrl modifier. Then on Blink, configure Ctrl to send `Esc on release`. That should give you both as you want them.

**Reply by d34dh0r53** [reply]
@carloscabanero I see what you’re saying, but since the Magic keyboard doesn’t have an ESC key it makes sense to have the Caps Lock remapped to ESC via iPadOS (for applications other than Blink).  I think Blink just needs an additional mapping for the ESC key that allows for Ctrl as a modifier.

**Reply by jfsicilia** [reply]
@carloscabanero I tried the settings you specify, they work perfectly on blink shell, but Ctrl won’t send `Esc on release` in VSCode, so can’t use the same key for ESC and CTRL.
